src/web/tools/EventManager.ts

```diff
@@ -1,186 +1,103 @@
+    // Events that correspond to removing the pointer (pointerup, touchend) have condition, that they don't perform any action when activePointersCounter
+    // is equal to 0. This prevents counter from going to negative values, when pointers are removed from view after one of them has been cancelled
   }
+  protected onPointerOutOfBounds(_event: AdaptedEvent): void {}
 
-  private onPointerDown(_event: AdaptedPointerEvent): void {}
-  private onPointerUp(_event: AdaptedPointerEvent): void {}
-  private onPointerMove(_event: AdaptedPointerEvent): void {}
-  private onPointerOut(_event: AdaptedPointerEvent): void {}
-  private onPointerEnter(_event: AdaptedPointerEvent): void {}
-  private onPointerCancel(_event: AdaptedPointerEvent): void {}
-  private onPointerOutOfBounds(_event: AdaptedPointerEvent): void {}
-
-  public setOnPointerDown(
-    callback: (event: AdaptedPointerEvent) => void
-  ): void {
+  public setOnPointerDown(callback: (event: AdaptedEvent) => void): void {
     this.onPointerDown = callback;
   }
-  public setOnPointerUp(callback: (event: AdaptedPointerEvent) => void): void {
+  public setOnPointerAdd(callback: (event: AdaptedEvent) => void): void {
+    this.onPointerAdd = callback;
+  }
+  public setOnPointerUp(callback: (event: AdaptedEvent) => void): void {
     this.onPointerUp = callback;
   }
-  public setOnPointerMove(
-    callback: (event: AdaptedPointerEvent) => void
-  ): void {
+  public setOnPointerRemove(callback: (event: AdaptedEvent) => void): void {
+    this.onPointerRemove = callback;
+  }
+  public setOnPointerMove(callback: (event: AdaptedEvent) => void): void {
     this.onPointerMove = callback;
   }
-  public setOnPointerOut(callback: (event: AdaptedPointerEvent) => void): void {
+  public setOnPointerOut(callback: (event: AdaptedEvent) => void): void {
     this.onPointerOut = callback;
   }
-  public setOnPointerEnter(
-    callback: (event: AdaptedPointerEvent) => void
-  ): void {
+  public setOnPointerEnter(callback: (event: AdaptedEvent) => void): void {
     this.onPointerEnter = callback;
   }
-  public setOnPointerCancel(
-    callback: (event: AdaptedPointerEvent) => void
-  ): void {
+  public setOnPointerCancel(callback: (event: AdaptedEvent) => void): void {
     this.onPointerCancel = callback;
   }
   public setOnPointerOutOfBounds(
-    callback: (event: AdaptedPointerEvent) => void
+    callback: (event: AdaptedEvent) => void
   ): void {
     this.onPointerOutOfBounds = callback;
   }
 
-  private mapEvent(
-    event: PointerEvent,
-    eventType: EventTypes
-  ): AdaptedPointerEvent {
-    return {
-      x: event.clientX,
-      y: event.clientY,
-      offsetX: event.offsetX,
-      offsetY: event.offsetY,
-      pointerId: event.pointerId,
-      eventType: eventType,
-      pointerType: event.pointerType,
-      buttons: event.buttons,
-      time: event.timeStamp,
-    };
-  }
-
   public isPointerInBounds({ x, y }: { x: number; y: number }): boolean {
```

Comment:
This function is also in the `GestureHandler`, maybe moving it out to some `utils` makes sense?